acceptance/tests/vault/vault_test.go

```diff
@@ -39,5 +44,89 @@ func TestVault_Create(t *testing.T) {
 	require.NoError(t, err)
 	logger.Log(t, "Auth List: ", authList)
 	require.NotNil(t, authList["kubernetes/"])
-	time.Sleep(time.Second * 60)
+}
+
+// Installs Vault, bootstraps it with secrets, policies, and Kube Auth Method
+// then creates a gossip encryption secret and uses this to bootstrap Consul.
+func TestVault_BootstrapConsulGossipEncryptionKey(t *testing.T) {
+	cfg := suite.Config()
+	ctx := suite.Environment().DefaultContext(t)
+
+	consulReleaseName := helpers.RandomName()
+	vaultReleaseName := helpers.RandomName()
+	consulClientServiceAccountName := fmt.Sprintf("%s-consul-client", consulReleaseName)
+	consulServerServiceAccountName := fmt.Sprintf("%s-consul-server", consulReleaseName)
+
+	vaultCluster := vault.NewVaultCluster(t, nil, ctx, cfg, vaultReleaseName)
+	vaultCluster.Create(t, ctx)
+	// Vault is now installed in the cluster.
+
+	// Now fetch the Vault client so we can create the policies and secrets.
+	vaultClient := vaultCluster.VaultClient(t)
+
+	// Create the Vault Policy for the gossip key.
+	logger.Log(t, "Creating the gossip policy")
+	rules := `
+path "consul/data/secret/gossip" {
+  capabilities = ["read"]
+}`
+	err := vaultClient.Sys().PutPolicy("consul-gossip", rules)
+	require.NoError(t, err)
+
+	// Create the Auth Roles for consul-server + consul-client.
+	logger.Log(t, "Creating the gossip auth roles")
+	params := map[string]interface{}{
+		"bound_service_account_names":      consulClientServiceAccountName,
+		"bound_service_account_namespaces": "default",
+		"policies":                         "consul-gossip",
+		"ttl":                              "24h",
+	}
+	_, err = vaultClient.Logical().Write("auth/kubernetes/role/consul-client", params)
+	require.NoError(t, err)
+
+	params = map[string]interface{}{
+		"bound_service_account_names":      consulServerServiceAccountName,
+		"bound_service_account_namespaces": "default",
+		"policies":                         "consul-gossip",
+		"ttl":                              "24h",
+	}
+	_, err = vaultClient.Logical().Write("auth/kubernetes/role/consul-server", params)
+	require.NoError(t, err)
+
```

Comment:
I will probably dry this up in the next PR for serverTLS as it will need to also create Policies and setup Roles